internal/pkg/action/operator_uninstall_test.go

```
@@ -0,0 +1,248 @@
+			},
+			Spec: apiextv1.CustomResourceDefinitionSpec{
+				Group: "etcd.database.coreos.com",
+				Names: apiextv1.CustomResourceDefinitionNames{
+					ListKind: "EtcdClusterList",
+				},
+			},
+		}
+		etcdcluster1 = &unstructured.Unstructured{}
+		etcdcluster1.SetGroupVersionKind(etcdclusterGVK)
+		etcdcluster1.SetNamespace("ns1")
+		etcdcluster1.SetName("cluster1")
+
+		etcdcluster2 = &unstructured.Unstructured{}
+		etcdcluster2.SetGroupVersionKind(etcdclusterGVK)
+		etcdcluster2.SetNamespace("ns2")
+		etcdcluster2.SetName("cluster2")
+
+		etcdcluster3 = &unstructured.Unstructured{}
+		etcdcluster3.SetGroupVersionKind(etcdclusterGVK)
+		// Empty namespace to simulate cluster-scoped object.
+		etcdcluster3.SetNamespace("")
+		etcdcluster3.SetName("cluster3")
+
+		cl := fake.NewClientBuilder().
+			WithObjects(operator, csv, og, sub, crd, etcdcluster1, etcdcluster2, etcdcluster3).
+			WithScheme(sch).
+			Build()
+		cfg.Scheme = sch
+		cfg.Client = cl
+		cfg.Namespace = "etcd-namespace"
+	})
+
+	It("should fail due to missing subscription", func() {
+		uninstaller := internalaction.NewOperatorUninstall(&cfg)
+		// switch to package without a subscription for it
+		uninstaller.Package = "redis"
+		err := uninstaller.Run(context.TODO())
+		Expect(err).To(MatchError(&internalaction.ErrPackageNotFound{PackageName: "redis"}))
+	})
+
+	It("should fail due to missing csv", func() {
+		// switch to missing csv
+		// this is not an error condition, we simply delete the subscription and exit
+		sub.Status.InstalledCSV = ""
+		Expect(cfg.Client.Update(context.TODO(), sub)).To(Succeed())
+
+		uninstaller := internalaction.NewOperatorUninstall(&cfg)
+		uninstaller.Package = etcd
+		err := uninstaller.Run(context.TODO())
+		Expect(err).To(BeNil())
+
+		subKey := types.NamespacedName{Name: "etcd-sub", Namespace: "etcd-namespace"}
+		s := &v1alpha1.Subscription{}
+		Expect(cfg.Client.Get(context.TODO(), subKey, s)).To(WithTransform(apierrors.IsNotFound, BeTrue()))
```

Can you bump the gomega dependency to get https://github.com/onsi/gomega/pull/423? Otherwise this will panic instead of failing when Get returns nil.